internal/operators/handler/handler.go

```diff
@@ -0,0 +1,132 @@
+	return restoperators.NewListSupportedOperatorsOK().
+		WithPayload(h.operatorsAPI.GetSupportedOperators())
+}
+
+// ListOfClusterOperators Lists operators to be monitored for a cluster.
+func (h *Handler) ListOfClusterOperators(ctx context.Context, params restoperators.ListOfClusterOperatorsParams) middleware.Responder {
+	operatorsList, err := h.GetMonitoredOperators(ctx, params.ClusterID, params.OperatorName, h.db)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewListOfClusterOperatorsOK().WithPayload(operatorsList)
+}
+
+// ReportMonitoredOperatorStatus Controller API to report of monitored operators.
+func (h *Handler) ReportMonitoredOperatorStatus(ctx context.Context, params restoperators.ReportMonitoredOperatorStatusParams) middleware.Responder {
+	err := h.UpdateMonitoredOperatorStatus(ctx, params.ClusterID, params.ReportParams.Name, params.ReportParams.Status, params.ReportParams.StatusInfo)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	return restoperators.NewReportMonitoredOperatorStatusOK()
+}
+
+// GetMonitoredOperators retrieves list of monitored operators for a cluster
+func (h *Handler) GetMonitoredOperators(ctx context.Context, clusterID strfmt.UUID, operatorName *string, db *gorm.DB) (models.MonitoredOperatorsList, error) {
+	log := logutil.FromContext(ctx, h.log)
+	var where *gorm.DB
+	if operatorName == nil || *operatorName == "" {
+		where = db.Where("cluster_id = ?", clusterID)
+	} else {
+		where = db.Where("name = ? and cluster_id = ?", *operatorName, clusterID)
+	}
+
+	var operatorsList = models.MonitoredOperatorsList{}
+	if err := where.Find(&operatorsList).Error; err != nil {
+		log.WithError(err).Errorf("failed to find monitored operators")
+		if gorm.IsRecordNotFoundError(err) {
+			return nil, common.NewApiError(http.StatusNotFound, err)
+		}
+	}
+
+	return operatorsList, nil
+}
+
+// UpdateMonitoredOperatorStatus updates status and status info of a monitored operator for a cluster
+func (h *Handler) UpdateMonitoredOperatorStatus(ctx context.Context, clusterID strfmt.UUID, monitoredOperatorName string, status models.OperatorStatus, statusInfo string) error {
+	log := logutil.FromContext(ctx, h.log)
+
+	txSuccess := false
+	tx := h.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("update monitored operator failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("update monitored operator failed")
+			tx.Rollback()
+		}
+	}()
+
+	monitoredOperators, err := h.GetMonitoredOperators(ctx, clusterID, &monitoredOperatorName, tx)
+	if err != nil {
+		return err
+	}
+	if len(monitoredOperators) == 0 {
+		log.WithField("clusterID", clusterID).WithField("operator", monitoredOperatorName).Infof("Monitored operator not found")
+		return common.NewApiError(http.StatusNotFound, errors.New("monitored operator not found"))
+	}
```

Comment:
I think it's possible to send here `monitoredOperatorName = ""` which would mean there would be returned multiple `monitoredOperators` and we update the first one. Maybe would be better to return error in case `len(monitoredOperators) > 1` ?

Reply:
WDYT about rejecting `monitoredOperatorName = ""`? This way we won't need to query the DB.

Reply:
Honestly I don't know `gorm`, so I can't think of any corner case where this could happen other than with `""`, so I think solving it as you suggest would be better.

Reply:
Done